File: processor/src/test/resources/approvals/jpa/OrderDateAttributeConverter.java
package test.boundary.persistence.jpa;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;
import java.time.LocalDate;
import javax.annotation.processing.Generated;
import scenarios.java.OrderDate;

@Generated("com.qualityminds.lazyval.processor.internal.codegen.JpaGenerator")
@Converter(
    autoApply = true
)
public class OrderDateAttributeConverter implements AttributeConverter<OrderDate, LocalDate> {
  public LocalDate convertToDatabaseColumn(OrderDate type) {
    if(type == null) {
      return null;
    }
    return type.value();
  }

  public OrderDate convertToEntityAttribute(LocalDate dbValue) {
    if(dbValue == null) {
      return null;
    }
    return new OrderDate(dbValue);
  }
}

File: ksp/src/test/resources/approvals/jpa/OrderDateAttributeConverter.kt
package test.boundary.persistence.jpa

import jakarta.`annotation`.Generated
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import java.time.LocalDate
import scenarios.kotlin.OrderDate

@Generated("com.qualityminds.lazyval.ksp.internal.codegen.JpaGenerator")
@Converter(autoApply = true)
public class OrderDateAttributeConverter : AttributeConverter<OrderDate?, LocalDate?> {
  override fun convertToDatabaseColumn(type: OrderDate?): LocalDate? = type?.date

  override fun convertToEntityAttribute(dbValue: LocalDate?): OrderDate? = dbValue?.let { OrderDate(dbValue) }
}
